src/Engine/Utils.java

```diff
@@ -381,9 +396,9 @@ public int compare(SearchNode o1, SearchNode o2)
       int firstDist = Math.abs(o1.x - xDest) + Math.abs(o1.y - yDest);
       int secondDist = Math.abs(o2.x - xDest) + Math.abs(o2.y - yDest);
 
-      int firstPow = movesLeftGrid[o1.x][o1.y] - ((hasDestination) ? firstDist : 0);
-      int secondPow = movesLeftGrid[o2.x][o2.y] - ((hasDestination) ? secondDist : 0);
-      return secondPow - firstPow;
+      int firstPow = costGrid[o1.x][o1.y] - ((hasDestination) ? firstDist : 0);
+      int secondPow = costGrid[o2.x][o2.y] - ((hasDestination) ? secondDist : 0);
```

Comment:
I believe the distances should now be *added* to the cost, and the variable names clarified.

Reply:
Yes